rdagent/core/utils.py

```diff
@@ -119,10 +118,9 @@ def multiprocessing_wrapper(func_calls: list[tuple[Callable, tuple]], n: int) ->
     """
     if n == 1:
         return [f(*args) for f, args in func_calls]
-
-    with mp.Pool(processes=n) as pool:
+    with mp.Pool(processes=max(1, min(n, len(func_calls)))) as pool:
         results = [
-            pool.apply_async(_subprocess_wrapper(f, APIBackend.cache_seed_gen.get_next_seed()), args=args)
+            pool.apply_async(_subprocess_wrapper, args=(f, random.randint(0, 10000), args)) # noqa: S311
```

Comment:
Please use  get_next_seed;
We may use other random seed mechanism if we want to solve the confliction with random.

Reply:
Why should we use `# noqa: S311`?

Reply:
[S311](https://docs.astral.sh/ruff/rules/suspicious-non-cryptographic-random-usage/), i don't think we need this encryption requirements

Reply:
> Please use get_next_seed; We may use other random seed mechanism if we want to solve the confliction with random.

APIBackend cannot be called here because a circular import occurs

Reply:
Import  it in the function to avoid circular import